docs/tutorial/tests/steps/library_steps.py

```diff
@@ -1,34 +1,72 @@
-from helper_methods.library_catalog import Catalog
-from helper_methods.verification_helper_methods import verify_returned_books
-from pytest import fixture
-from pytest_bdd_ng import given, parsers, step, then, when
+import re
+from typing import Literal
 
+from src.catalog import Book, Catalog
 
-@fixture
-def context()
-    """ Create a placeholder object to use in place of Cucumber's context object. The context object allows us to pass state between steps. """
-    class dummy():
-        pass
+from messages import DataTable, Step  # type:ignore[attr-defined]
+from pytest_bdd import given, step, then, when
 
 
-    return dummy()
+def get_books_from_data_table(data_table: DataTable):
+    # Gherkin data-tables have no title row by default, but we could use if we want.
+    step_data_table_titles = [cell.value for cell in data_table.rows[0].cells]
+    assert step_data_table_titles == ["Author", "Title"]
 
+    author_and_title_list = [[cell.value for cell in row.cells] for row in data_table.rows[1:]]
```

Comment:
List comprehensions are much more elegant than a 'while' loop, but I think that the 'while' loop can be understood even by someone who doesn't know Python.

Reply:
> List comprehensions are much more elegant than a 'while' loop, but I think that the 'while' loop can be understood even by someone who doesn't know Python.

So, I agree that `while` and `for` are more familiar to folks coming from other languages, _and_ I would like us to encourage modern python as much as possible.

We may want to point out in code examples:
> "... since this tutorial assumes experience with other languages, we will use the more familiar `while` loop. Going forward, expect to see the more pythonic use of list comprehensions instead"